modules/openideal_user/src/Plugin/RulesAction/FlagAction.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Drupal\openideal_user\Plugin\RulesAction;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\flag\FlagServiceInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'Save entity' action.
```

Comment:
Good stuff. Can we publish this to flag module issues queue to add `Rules` support to the module?

Reply:
Reworked the module to be more "flexible" without hardcoded details. Please, take a look.

Reply:
It looks better, but we'll have to adjust some details before publishing it for review.
We'll get back to this later.